frontend/src/components/MeetingRoom/PopupMenuToggleButton/PopupMenuToggleButton.tsx

```diff
@@ -0,0 +1,93 @@
+import { ReactElement, useRef, useState } from 'react';
+import { Grow, Paper, Popper, Tooltip } from '@mui/material';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import { ClickAwayListener, PopperChildrenProps } from '@mui/base';
+import ToolbarButton from '../ToolbarButton';
+import ArchivingToggle from '../ArchivingToggle';
+import EmojiGrid from '../EmojiGrid';
+import LayoutToggleButton from '../LayoutToggleButton';
+import useSessionContext from '../../../hooks/useSessionContext';
+
+/**
+ * PopupMenuToggleButton Component
+ *
+ * Displays a clickable button that opens a grid of hidden toolbar buttons for smaller viewport devices.
+ * @returns {ReactElement} - The PopupMenuToggleButton Component.
+ */
+const PopupMenuToggleButton = (): ReactElement => {
+  const { subscriberWrappers } = useSessionContext();
+  const anchorRef = useRef<HTMLButtonElement>(null);
+  const [open, setOpen] = useState<boolean>(false);
+  const isViewingScreenShare = subscriberWrappers.some((subWrapper) => subWrapper.isScreenshare);
+  const [openEmojiGridMobile, setOpenEmojiGridMobile] = useState<boolean>(true);
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleToggle = () => {
+    setOpen((prevOpen) => !prevOpen);
+  };
+
+  return (
+    <>
+      <Tooltip
+        title="Access additional toolbar items"
+        aria-label="open additional toolbar items menu"
+      >
+        <ToolbarButton
+          data-testid="hidden-toolbar-items"
+          onClick={handleToggle}
+          icon={<MoreVertIcon style={{ color: `${!open ? 'white' : 'rgb(138, 180, 248)'}` }} />}
+          ref={anchorRef}
+        />
+      </Tooltip>
+
+      <Popper open={open} anchorEl={anchorRef.current} transition disablePortal placement="bottom">
+        {({ TransitionProps, placement }: PopperChildrenProps) => (
+          <Grow
+            {...TransitionProps}
+            style={{
```

Comment:
D'you think we could separate the button and the menu itself into different component files?
Even if it's rendered here still